### PR TITLE
Handle cupy array in `Dataframe.__setitem__`

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #7337 

This PR introduces changes to handle `cupy` array in `DataFrame.__setitem__`.

### CODE REVIEW

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -777,7 +777,16 @@ def __setitem__(self, arg, value):
                     self.insert(len(self._data), arg, value)
 
         elif isinstance(
-            arg, (list, np.ndarray, pd.Series, Series, Index, pd.Index)
+            arg,
+            (
+                list,
+                np.ndarray,
+                cupy.ndarray,
```
Comment (shwina) [inline python/cudf/cudf/core/dataframe.py:784]: Should we generally handle objects with `__cuda_array_interface__` here?

Comment (galipremsagar) [reply python/cudf/cudf/core/dataframe.py]: Yup, we have generic utility `is_column_like` for this. I changed it to `can_convert_to_column` to also handle list-like inputs.

**Review by shwina — APPROVED** [review]
LGTM!

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**